src/tools/ustore_cli.cpp

```diff
@@ -0,0 +1,423 @@
+           value("collection name", name)) |
+          (required("drop").set(action, std::string("drop")) & required("--name") & value("collection name", name)) |
+          required("list").set(action, std::string("list")) |
+          ((required("import").set(action, std::string("import")) &
+            (required("--input") & value("input", input_file)).doc("Input file path") &
+            (required("--id") & value("id field", id_field)).doc("The field which data will use as key(s)")) |
+           (required("export").set(action, std::string("export")) &
+            (required("--output") & value("output", output_ext)).doc("Output file path"))) &
+              ((required("--mlimit") & value("memory limit", memory_limit))
+                   .doc("Size of available RAM for a specific operation in bytes"),
+               (option("--name") & value("collection name", name)))));
+
+    auto snapshot = (option("snapshot").set(db_object, std::string("snapshot")) &
+                     ((required("create").set(action, std::string("create"))) |
+                      (required("export").set(action, std::string("export")) & value("path", export_path)) |
+                      (required("drop").set(action, std::string("drop")) & value("snapshot id", snap_id)) |
+                      (required("list").set(action, std::string("list")))));
+
+    auto cli = ((required("--url") & value("URL", url)).doc("Server URL"),
+                (collection | snapshot),
+                option("-h", "--help").set(help).doc("Print this help information on this tool and exit"));
+
+    if (!parse(argc, argv, cli)) {
+        std::cerr << make_man_page(cli, argv[0]);
+        return 1;
+    }
+    if (help) {
+        std::cout << make_man_page(cli, argv[0]);
+        return 0;
+    }
+
+    database_t db;
+    db.open(url.c_str()).throw_unhandled();
+
+    if (db_object == "collection") {
+        if (action == "create")
+            collection_create(db, name);
+        else if (action == "drop")
+            collection_drop(db, name);
+        else if (action == "list")
+            collection_list(db);
+        else if (action == "import")
+            docs_import(db, name, input_file, id_field, memory_limit);
+        else if (action == "export")
+            docs_export(db, name, output_ext, memory_limit);
+        return 0;
+    }
+    else if (db_object == "snapshot") {
+        if (action == "create")
+            snapshot_create(db);
+        else if (action == "export")
+            snapshot_export(db, export_path);
+        else if (action == "drop")
+            snapshot_drop(db, snap_id);
+        else if (action == "list")
+            snapshot_list(db);
+        return 0;
+    }
+
+    std::string input;
+    std::vector<std::string> commands;
+    std::regex const reg_exp(" +(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
+    std::regex_token_iterator<std::string::iterator> const end_tokens;
+
+    while (true) {
```

Comment:
This would have been much better structured if the internals of the Read-Evaluate-Print-Loop were a separate function without his many layers of nesting, `if`-s and `break`-s.